include/boost/json/impl/object.ipp

```diff
@@ -270,6 +274,11 @@ object(detail::unchecked_object&& uo)
             }
 
             // handle duplicate
+            if( !uo.ignore_duplicate_keys() )
```

Comment:
This is a constant in the loop, so you should lift the conditional out of the loop